src/vivarium_gates_iv_iron/components/pregnancy.py

```diff
@@ -98,65 +132,111 @@ def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+            pop.index)
+
+        prepostpartum_ends_this_step = (
+           (
+               (pop['pregnancy_status'] == models.MATERNAL_DISORDER_STATE)
+               | (pop['pregnancy_status'] == models.NO_MATERNAL_DISORDER_STATE)
+               & (event.time - pop["pregnancy_state_change_date"] >
+                  pd.Timedelta(days=PREPOSTPARTUM_DURATION_DAYS))  # One time step
+            )
+        )
+        postpartum_ends_this_step = (
+                (pop['pregnancy_status'] == models.POSTPARTUM_STATE)
+                & (event.time - pop["pregnancy_state_change_date"] > pd.Timedelta(days=POSTPARTUM_DURATION_DAYS))
+        )
+
+        # Determine who dies
+        maternal_disorder_death_draw = self.randomness.get_draw(pop.index, additional_key="maternal_disorder_death")
+        would_die_due_to_maternal_disorders = maternal_disorder_death_draw < self.probability_maternal_deaths(pop.index)
+        died_due_to_maternal_disorders = pregnancy_ends_this_step & would_die_due_to_maternal_disorders
+        died_due_to_background_causes_index = self.randomness.filter_for_rate(pop.index, rate=self.background_mortality_rate(pop.index),
+                                                                        additional_key="other_cause_death")
+        died_due_to_background_causes = pd.Series(False, index=pop.index)
+        died_due_to_background_causes.loc[died_due_to_background_causes_index] = True
+        died_due_to_background_causes.loc[died_due_to_maternal_disorders] = False
+        died_this_step = died_due_to_maternal_disorders | died_due_to_background_causes
+
+        pop.loc[died_this_step, "alive"] = "dead"
+        pop.loc[died_this_step, "exit_time"] = event.time
+        pop.loc[died_this_step, "years_of_life_lost"] = self.life_expectancy(pop.loc[died_this_step].index)
+        pop.loc[died_due_to_maternal_disorders, "cause_of_death"] = "maternal_disorders"
+        pop.loc[died_due_to_background_causes, "cause_of_death"] = "other_causes"
+
+        # Update new pregnancies
+        # TODO: If you want to be mutually exclusive from death make this
+        # pregnant_this_step = pregnant_this_step & ~died_this_step
+        pop.loc[pregnant_this_step, "pregnancy_status"] = models.PREGNANT_STATE
+        pop.loc[pregnant_this_step, "pregnancy_outcome"] = pregnancy_outcome.loc[pregnant_this_step]
+        pop.loc[pregnant_this_step, "sex_of_child"] = sex_of_child.loc[pregnant_this_step]
+        pop.loc[pregnant_this_step, "birth_weight"] = birth_weight.loc[pregnant_this_step]
+        pop.loc[pregnant_this_step, "pregnancy_duration"] = pregnancy_duration
+        pop.loc[pregnant_this_step, "pregnancy_state_change_date"] = event.time
+
+        # Pregnancy to maternal disorder state and no maternal disorder state
+        maternal_disorder_this_step = maternal_disorder_this_step & pregnancy_ends_this_step
+        no_maternal_disorder_this_step = pregnancy_ends_this_step & ~maternal_disorder_this_step
```

Comment:
It's weird that you flipped the order of these.

Reply:
Weird, but immaterial. There's a lot of cosmetic changes we ought to make.